telink: tlx: apply ieee802154 optimization to zephyr v3.3

damien0x0023 opened the pull request:
- add config IEEE802154_TLX_OPTIMIZATION (n by default)
- modify radio reset
based on de86a258732b52709d8b4f701a55e00304bf72ad

damien0x0023 committed:
telink: tlx: apply tlx optimization

add config IEEE802154_TLX_OPTIMIZATION
modify radio reset
based on de86a258732b52709d8b4f701a55e00304bf72ad

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

pulin1103 commented on drivers/ieee802154/ieee802154_tlx.c:
```diff
@@ -1130,11 +1153,6 @@ static int tlx_tx(const struct device *dev,
 
 	do {
 
-		if (net_pkt_ieee802154_mac_hdr_rdy(pkt)) {
```
@damien0x0023 This optimization was previously added to the develop branch to prevent redundant encryption at the radio layer and Line 1133-1137 should not be removed.

damien0x0023 replied:
Will revert it.

haiwentelink commented on drivers/ieee802154/ieee802154_tlx.c:
```diff
@@ -59,6 +60,10 @@ static struct tlx_enh_ack_table enh_ack_table;
 
+#if CONFIG_SOC_SERIES_RISCV_TELINK_TLX_RETENTION
+__attribute__((section(".bss")))  uint8_t rxpkt_buffer[TLX_TRX_LENGTH] = {0};
+__attribute__((section(".bss")))  uint8_t txpkt_buffer[TLX_TRX_LENGTH] = {0};
```
bss is locate in dlm which will not hold ram in retention mode , so need to check it .

damien0x0023 replied:
@haiwentelink has confirmed with @liang4804 that we do not need to hold them in the ilm. 
The data inside tx_buffer/rx_buffer can be dropped during deep sleep with the retention period.

haiwentelink reviewed (changes requested):
need to check

damien0x0023 committed:
telink: tlx: avoid duplicate encrypiton

add net_pkt_ieee802154_mac_hdr_rdy back
based on d5431889e0d16f9040f003c6f2ada6d57ec81d53.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

liucoldstar commented on drivers/ieee802154/Kconfig.tlx:
```diff
@@ -48,6 +48,13 @@ config IEEE802154_TLX_DELAY_TRX_ACC
 	  Accuracy of the clock used for scheduling radio delayed operations (delayed transmission
 	  or delayed reception), in ppm.
 
+config IEEE802154_TLX_OPTIMIZATION
```
config IEEE802154_TLX_OPTIMIZATION
	bool "Optimize 802.15.4 RF performance for TLX SoCs"
	default n
	help
	  Improve RF performance in IEEE 802.15.4 communication on TLX SoCs.

damien0x0023 replied:
Updated.

damien0x0023 committed:
telink: tlx: improve description

and code format.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

damien0x0023 commented:
Forgot to choose Squash and Merge for this PR.
I thus squash these three commits into one commit 86fcdb43b64d6d10ad81172c61495c9e40de0403, and perform a force push to develop_v3.3.
